partner_address_format/tests/test_get_address.py

```diff
@@ -0,0 +1,124 @@
+        }
+        view = self.contact_use_parent.with_context(ctx).pick_view()
+        self.assertEqual(self.view_wired,
+                         view,
+                         'Returns:  %s' % view.name)
+
+    def test_005_pick_view_default(self):
+        """Forcing the default view if nothing properly configured"""
+        self.contact_use_parent.country_id.address_format_id = False
+        self.contact_use_parent.country_id = False
+        self.main_company.address_format_id = False
+        view = self.contact_use_parent.pick_view()
+        self.assertEqual(self.view_default, view,
+                         'Expecting the default view')
+
+    def test_010_rendering_ok(self):
+        """Rendering properly a template"""
+        self.contact_use_parent.country_id = self.mexico
+        view = self.contact_use_parent.get_address_from_template()
+        self.assertIn('Mexico', view,
+                      'Getting the rendering of a template fails.')
+
+    def test_011_rendering_not_ok(self):
+        """Rendering properly a broken template"""
+        ctx = {'view_id': 'partner_address_format.demo_broken_format'}
+        html = self.contact_use_parent.with_context(ctx).get_address_from_template()  # noqa
+        self.assertIn('broken_element',
+                      html,
+                      'Element broken returns an unexpected output')
+
+    def by_template_id(self, xml_id):
+        """Assuming all properly configured, check the templates are working
+
+        This set of tests are supposed to be called in every module that create
+        a new template for address in order to test the template itself.
+        In this case  we will be testing the template for the address added in
+        this module with all known cases where the partner can be set.
+
+        Remember, we can have logic inside the template to render it this
+        method is to generalize the test It is more or less a framework of
+        tests with the minimal required type of partners to be rendered."""
+        ctx = {
+            'view_id': xml_id
+        }
+        # Normal Contact which use a parent address from the company
+        html = self.contact_use_parent.with_context(
+            ctx).get_address_from_template()
+        self.assertNotIn('QWebException', html,
+                         'Template: %s  .' % xml_id)
+        html = self.contact_no_company.with_context(
+            ctx).get_address_from_template()
+        self.assertNotIn('QWebException', html,
+                         'Template: %s  .' % xml_id)
+        html = self.company_no_parent.with_context(
+            ctx).get_address_from_template()
+        self.assertNotIn('QWebException', html,
+                         'Template: %s  .' % xml_id)
+
+    def test_020_address_templates(self):
+        """Testing all templates on this module."""
+        [self.by_template_id(template) for template in self.templates]
```

Comment:
Asserts are generic ones, this method is to trigger a test per template (asserts into by_template_id) do you think there is a more ellegant way to achieve this?